pythonforandroid/bootstrap.py

```diff
@@ -299,9 +363,26 @@ def fry_eggs(self, sitepackages):
                 shprint(sh.rm, '-rf', d)
 
 
-def expand_dependencies(recipes):
+def expand_dependencies(recipes, ctx):
+    """ This function expands to lists of all different available
+        alternative recipe combinations, with the dependencies added in
+        ONLY for all the not-with-alternative recipes.
+        (So this is like the deps graph very simplified and incomplete, but
+         hopefully good enough for most basic bootstrap compatibility checks)
+    """
+
+    # Add in all the deps of recipes where there is no alternative:
+    recipes_with_deps = list(recipes)
+    for entry in recipes:
+        if not isinstance(entry, (tuple, list)) or len(entry) == 1:
+            if isinstance(entry, (tuple, list)):
+                entry = entry[0]
```

Comment:
There's something fishy here :fish: :smile: 
First it doesn't seem covered, second it means `expand_dependencies()` would be a bit polymorphic in a way which can lead to unexpected behavior or code not so easy to follow.

Reply:
What do you mean with polymorphic? This is the usual behavior all graph lists work, in fact I did not introduce this at all but just stuck to it as it already behaved (see the low loop.) So no, this is as it should be. (Dependencies can be a string or a tuple of alternatives, that is how things are used everywhere)

The coverage is a good point though, let me look into that

Reply:
Thanks for the reply. Great that you will look at the coverage, who knows maybe we will come across another old bug or dead code :smile: